[Docs] Adding useQuery related information (#6231)
PipedreamHQ/pipedream: docs/docs/components/api/README.md

| `description`    | `string`                             | optional  | Displayed near the prop input. Typically used to contextualize the prop or provide instructions to help users input the correct value. Markdown is supported.                                                                                                                                                                                                                                                                                                                  |
| `options`        | `string[]` or `object[]` or `method` | optional  | Provide an array to display options to a user in a drop down menu.<br>&nbsp;<br>**`[]` Basic usage**<br>Array of strings. E.g.,<br>`['option 1', 'option 2']`<br>&nbsp;<br>**`object[]` Define Label and Value**<br>`[{ label: 'Label 1', value: 'label1'}, { label: 'Label 2', value: 'label2'}]`<br>&nbsp;<br>**`method` Dynamic Options**<br>You can generate options dynamically (e.g., based on real-time API requests with pagination). See configuration details below. |

> | `useQuery`    | `boolean`                             | optional  | Use in conjunction with **Dynamic Options**. If set to `true`, the prop accepts a real-time query that can be used by the `options` method to obtain results according to that query.                                                                                                                                                                                                                                                                                                                   |

Reviewer: ```suggestion
| `useQuery`    | `boolean`                             | optional  | Use in conjunction with **Dynamic Options**. If set to `true`, the prop accepts a real-time query that can be used by the `options` method to obtain results according to that query. 
```

Reviewer: Removed extra spaces at the end of the line.

> | `useQuery`    | `boolean`                             | optional  | Use in conjunction with **Dynamic Options**. If set to `true`, the prop will accept a real-time query that can be used by the `options` method to obtain results according to that query.                                                                                                                                                                                                                                                                                                                   |

Reviewer: ```suggestion
| `useQuery`    | `boolean`                             | optional  | Use in conjunction with **Dynamic Options**. If set to `true`, the prop accepts a real-time query that can be used by the `options` method to obtain results according to that query.                                                                                                                                                                                                                                                                                                                   |
```

Reviewer: Changed 'will accept' to 'accepts' for consistency with Google's Developer docs style guide (use present tense).

| ------------- | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `options()`   | `method`  | optional  | Typically returns an array of values matching the prop type (e.g., `string`) or an array of object that define the `label` and `value` for each option. The `page` and `prevContext` input parameter names are reserved for pagination (see below).<br>&nbsp;<br>When using `prevContext` for pagination, it must return an object with an `options` array and a `context` object with a `nextPageToken` key. E.g., `{ options, context: { nextPageToken }, }` |

> | `page`        | `integer` | optional  | Returns a `0` indexed page number. Use with APIs that accept a numeric page number for pagination.                                                                                                                                                                                                                                                                                                                                                         |

Reviewer: ```suggestion
| `page`        | `integer` | optional  | Returns a `0` indexed page number. Use with APIs that accept a numeric page number for pagination. 
```

Reviewer: Removed extra spaces at the end of the line.

> | `prevContext` | `string`  | optional  | Returns a string representing the context for the previous `options` execution. For use with APIs that accept a token representing the last record for pagination.                                                                                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| `prevContext` | `string`  | optional  | Returns a string representing the context for the previous `options` execution. Use with APIs that accept a token representing the last record for pagination.                                                                                                                                                                                                                                                                                              |
```

Reviewer: Removed 'For use with' and replaced with 'Use with' for consistency with Google's Developer docs style guide (use active voice).

> | `prevContext` | `string`  | optional  | Returns a string representing the context for the previous `options` execution. For use with APIs that accept a token representing the last record for pagination.                                                                                                                                                                                                                                                                                              |

Author: I only added an `s` on this line, so that the parameter `returns` a string (the same way the previous lines are worded)

> | `prevContext` | `string`  | optional  | Returns a string representing the context for the previous `options` execution. Use with APIs that accept a token representing the last record for pagination.                                                                                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| `prevContext` | `string`  | optional  | Returns a string representing the context for the previous `options` execution. Use with APIs that accept a token representing the last record for pagination. 
```

Reviewer: Removed extra spaces at the end of the line.

> | `query` | `string`  | optional  | Returns a string with the user input if the prop has the `useQuery` property set to `true`. For use with APIs that return items based on a query or search parameter.                                                                                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| `query` | `string`  | optional  | Returns a string with the user input if the prop has the `useQuery` property set to `true`. Use with APIs that return items based on a query or search parameter.                                                                                                                                                                                                                                                                                              |
```

Reviewer: Removed 'For use with' and replaced with 'Use with' for consistency with Google's Developer docs style guide (use active voice).

> | `query` | `string`  | optional  | Returns a string with the user input if the prop has the `useQuery` property set to `true`. Use with APIs that return items based on a query or search parameter.                                                                                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| `query` | `string`  | optional  | Returns a string with the user input if the prop has the `useQuery` property set to `true`. Use with APIs that return items based on a query or search parameter. 
```

Reviewer: Removed extra spaces at the end of the line.

### Diagram

> ![image-20200819210516311](https://res.cloudinary.com/pipedreamin/image/upload/v1683089643/d0iiggokfkwnmt4kckb5.png)

Reviewer: ```suggestion
![Image from August 19, 2020](https://res.cloudinary.com/pipedreamin/image/upload/v1683089643/d0iiggokfkwnmt4kckb5.png)
```

Reviewer: Changed image alt text to be more descriptive, Used proper date format
